packages/eas-cli/src/commands/build/configure.ts

```diff
@@ -15,15 +16,26 @@ export default class BuildConfigure extends Command {
       options: ['android', 'ios', 'all'],
       default: 'all',
     }),
+    'allow-experimental': flags.boolean({
+      description: 'Enable experimental configuration steps.',
+      default: false,
+    }),
   };
 
   async run() {
     const { flags } = this.parse(BuildConfigure);
     const platform = flags.platform as BuildCommandPlatform;
+    const allowExperimental = flags['allow-experimental'];
+    if (allowExperimental) {
+      log.warn(
+        'Project configuration will execute some additional steps, that might fail if structure of your native project is significantly different from "expo eject" or "expo init"'
```

Comment:
```suggestion
        'Project configuration will execute some additional steps that might fail if structure of your native project is significantly different from "expo eject" or "expo init"'
```

my only concern with this text is that it leaves users that don't have an app initialized this way without any understanding of how they should proceed if it does not work for them

Reply:
There could be two types of operations hidden under this flag (required and optional)
- for optional (like a renaming package) in case of error we can just abort process and print message to user to remove `--alow-experimental` flag
- for required values behavior would be the same as optional when running with the flag `--alow-experimental`, but if flag is not specified configure should throw descriptive error what needs to be configured for configure to work.

I didn't want to introduce that distinction in this PR because for now we only have one optional operation that needs it.